vault/build/titusvaults/sources/Vault.move

```diff
@@ -49,64 +74,133 @@ module titusvaults::Vault {
         withdrawer: address,
         amount: u64,
-        shares_burnt: u64,
+        shares_burnt: u64
     }
 
     #[event]
     struct StandardWithdrawVaultEvent has drop, store {
         withdrawer: address,
         amount: u64,
-        shares_burnt: u64,
+        shares_burnt: u64
     }
 
-    public entry fun set_current_time(_host: &signer) {
-        let host_addr = address_of(_host);
-        assert!(host_addr == @titusvaults, E_NOT_AUTHORIZED);
-        move_to(_host, Round {
-            initial_time: timestamp::now_microseconds(),
-            round: 0,
+    #[event]
+    struct RoundUpdatedEvent has drop, store {
+        new_round_id: u64
+    }
+
+    #[event]
+    struct StrikePriceUpdatedEvent has drop, store {
+        round_id: u64,
+        strike_price: u64
+    }
+
+    #[event]
+    struct PremiumPriceUpdatedEvent has drop, store {
+        round_id: u64,
+        premium_price: u64
+    }
+
+    #[event]
+    struct MintOptionsEvent has drop, store {
+        round_id: u64,
+        amount: u64,
+        strike_price: u64,
+        premium_price: u64,
+    }
+
+    // init round state
+    public entry fun initialize_round_state(_host: &signer) {
+        move_to(_host, RoundState {
+            current_round_id: 1,
+            shares: smart_table::new(),
+            strike_price: 0,
+            premium_price: 0,
+            round_start_time: 0,
+            deposit_start_time: 0,
+            option_creation_time: 0,
+            exercise_time: 0,
+            close_timestamp: 0,
+            total_amount_deposited: 0,
+            total_premium_collected: 0,
+            timestamps_set: false,
+            is_options_minted: false
         });
     }
+    
+    // keeper functions
+    public fun update_round_from_keeper(_host: &signer) acquires RoundState, VaultMap {
+        let host_addr = signer::address_of(_host);
+        let round_state = borrow_global_mut<RoundState>(@titusvaults);
+        let vault_map = borrow_global_mut<VaultMap>(@titusvaults);
+
+        let current_time = timestamp::now_microseconds();
+        let new_round = round_state.current_round_id + 1;
+
+        // Create a new RoundState for the new round (e.g., Round ID 2)
+        let new_round = RoundState {
+            current_round_id: new_round,
+            shares: smart_table::new(),
+            strike_price: round_state.strike_price,
+            premium_price: round_state.premium_price,
+            round_start_time: current_time,
+            deposit_start_time: current_time,
+            option_creation_time: current_time + DEPOSIT_PHASE_DURATION,
+            exercise_time: current_time + DEPOSIT_PHASE_DURATION + OPTION_EXPIRY_DURATION,
+            close_timestamp: current_time + DEPOSIT_PHASE_DURATION + OPTION_EXPIRY_DURATION + EXERCISE_BUFFER,
+            total_amount_deposited: round_state.total_amount_deposited,
+            total_premium_collected: round_state.total_premium_collected,
```

Comment:
same goes here. It should be zero, as in this round no premium has been collected yet.

Reply:
@0xriazaka 
Not yet resolved